### ISSUE TITLE
feat: assess single-model multi-domain inference via adapter hot-swap

### ISSUE THREAD

**Issue by NguyenJus** [NguyenJus/custom-sam-peft#118]
## Motivation

PEFT-based fine-tuning produces small per-domain adapter weights on top of a shared frozen SAM backbone. Rather than loading N full models for N domains (natural / medical / satellite / …), we should be able to keep **one** base model resident in VRAM and **swap adapter layers in/out** at request time. This is the natural payoff of going adapter-only.

## Goal

Assess whether — and how — to support a workflow like:

```
predictor = MultiDomainPredictor(base="sam3", adapters={
    "medical":   "ckpt/medical-lora",
    "satellite": "ckpt/satellite-qlora",
    "natural":   "ckpt/natural-lora",
})
predictor.predict(image, domain="medical")
predictor.predict(image, domain="satellite")  # swap, no base reload
```

## Questions to answer

- **Swap mechanics.** What's the fastest correct way to swap adapter state on a loaded `peft` model? (`set_adapter`, `load_adapter`, manual state_dict overlay, …) Overhead per swap?
- **Coverage across PEFT methods.** Does the answer differ for LoRA vs QLoRA vs other methods we support? (cf. #100 registry rework)
- **Mixed batches.** Can/should we route a batch with multiple domains through a single forward pass, or is one-domain-per-batch the right constraint?
- **Storage format.** What ships as a "domain adapter" artifact — bare adapter weights, or a small bundle with task config (class list, loss preset from #112, postprocess thresholds)?
- **API surface.** New CLI subcommand (`predict --domain X`) vs Python-only API vs config-driven. Consistency with existing eval/predict flow.
- **Failure modes.** Adapter trained against a different base revision; adapter target-modules mismatch; class-count mismatch on the head.
- **VRAM accounting.** Do all adapters need to live resident, or load-on-demand? What's the realistic upper bound on N?

## Adjacent / related

- #112 (domain-aware loss presets) — same domain taxonomy is useful here
- #100 (registry-driven PEFT methods) — affects how generically we can swap
- #98 (QLoRA checkpoint load from disk) — prerequisite for QLoRA adapters in this flow

## Out of scope for this issue

Actual implementation. This is the design/feasibility assessment; a follow-up feat issue captures the build once we know what we're building.

**Comment by NguyenJus** [comment]
OBE. This is on the users, not the package. This package is meant to facilitate rapid research given data, not supporting users in compiling their modalities together.
